modules/internal/ArrayViewSlice.chpl

```diff
@@ -0,0 +1,391 @@
+    }
+
+
+    //
+    // locality-oriented queries
+    //
+
+    proc dsiTargetLocales() {
+      return arr.dsiTargetLocales();
+    }
+
+    proc dsiHasSingleLocalSubdomain() param
+      return privDom.dsiHasSingleLocalSubdomain();
+
+    proc dsiLocalSubdomain() {
+      return privDom.dsiLocalSubdomain();
+    }
+
+    proc dsiNoFluffView() {
+      if canResolveMethod(arr, "dsiNoFluffView") {
+        return arr.dsiNoFluffView();
+      } else {
+        compilerError("noFluffView is not supported on this array type.");
+      }
+    }
+
+
+    //
+    // privatization
+    //
+
+    // Don't want to privatize a DefaultRectangular, so pass the query on to
+    // the wrapped array
+    proc dsiSupportsPrivatization() param
+      return _ArrInstance.dsiSupportsPrivatization();
+
+    proc dsiGetPrivatizeData() {
+      return (_DomPid, dom, _ArrPid, _ArrInstance);
+    }
+
+    proc dsiPrivatize(privatizeData) {
+      return new ArrayViewSliceArr(eltType=this.eltType,
+                                   _DomPid=privatizeData(1),
+                                   dom=privatizeData(2),
+                                   _ArrPid=privatizeData(3),
+                                   _ArrInstance=privatizeData(4));
+    }
+
+
+    //
+    // bulk-transfer
+    //
+
+    proc dsiSupportsBulkTransfer() param {
+      return arr.dsiSupportsBulkTransfer();
+    }
+
+    proc dsiSupportsBulkTransferInterface() param
+      return arr.dsiSupportsBulkTransferInterface();
+
+    proc _viewHelper(dims) {
+      compilerError("viewHelper not supported on ArrayViewSlice.");
+    }
+
+    proc _getViewDom() {
+      if _containsRCRE() {
+        // Rank-changes and reindexes know how to deal with nested rank-changes
+        // and reindexes, so hand off our indices and let them handle the rest.
+        var nextView = _getRCREView();
+        return nextView._viewHelper(dom.dsiDims());
+      } else {
+        return {(...dom.dsiDims())};
+      }
+    }
+
+    // contiguous transfer support
+    proc doiUseBulkTransfer(B) {
+      return arr.doiUseBulkTransfer(B);
+    }
+
+    proc doiCanBulkTransfer(viewDom) {
+      return arr.doiCanBulkTransfer(viewDom);
+    }
+
+    proc doiBulkTransfer(B, viewDom) {
+      arr.doiBulkTransfer(B, viewDom);
+    }
+
+    // strided transfer support
+    proc doiUseBulkTransferStride(B) {
+      return arr.doiUseBulkTransferStride(B);
+    }
+
+    proc doiCanBulkTransferStride(viewDom) {
+      return arr.doiCanBulkTransferStride(viewDom);
+    }
+
+    proc doiBulkTransferStride(B, viewDom) {
+      arr.doiBulkTransferStride(B, viewDom);
+    }
+
+    // distributed transfer support
+    proc doiBulkTransferToDR(B, viewDom) {
+      arr.doiBulkTransferToDR(B, viewDom);
+    }
+
+    proc doiBulkTransferFromDR(B, viewDom) {
+      arr.doiBulkTransferFromDR(B, viewDom);
+    }
+
+    proc doiBulkTransferFrom(B, viewDom) {
+      arr.doiBulkTransferFrom(B, viewDom);
+    }
```

Comment:
There are like 20 methods here that forward to arr. Is this a use case for delegation/forwarding?